### PR TITLE
async env and local frontend

### PR DESCRIPTION
**Pull request by jkminneti**
```
tuber dev-gql-server (appName)
```
no env vars. 

it pulls the input app from the remote tuber into your local database. it also does _local user impersonation_.

oh and also app env is async, lauren did frontend

wild friday night

### CODE REVIEW

File: pkg/k8s/configResource.go [hunk 1]
```diff
@@ -75,3 +75,36 @@ func GetConfigResource(name string, namespace string, kind string) (config *Conf
 
 	return
 }
+
+// GetConfigResourceWithToken is a step toward a larger refactor
```
Comment (jkminneti) [inline pkg/k8s/configResource.go:79]: killed this in the subsequent pr, #262

File: pkg/adminserver/web/pages/apps/[id].tsx [hunk 2]
```diff
@@ -227,13 +228,15 @@ const ShowApp = () => {
...
-					useSet={useSetAppEnvMutation}
-					useUnset={useUnsetAppEnvMutation}
-				/>
-			</Collapsible>
+			<CollapsedAsyncTextInputGroup 
+				tableHeading={'Environment Variables'} 
+				appName={app.name}
+				queryDocument={GetAppEnvDocument}
+				queryVars={{ name: app.name }}
+				queryData={function(data) { return data.getAppEnv}}
```
Comment (jkminneti) [inline pkg/adminserver/web/pages/apps/[id].tsx:236]: is this an absolute abuse of the any type? probably

Comment (Hbbb) [reply pkg/adminserver/web/pages/apps/[id].tsx:236]: Does this even need to exist? Why can't you just access `data.getAppEnv` directly inside of `CollapsedAsyncTextInputGroup`?

Comment (Hbbb) [reply pkg/adminserver/web/pages/apps/[id].tsx:236]: Actually I take that back. The fact that the accessor is a function makes this more generic. Nice!

**Review by Hbbb — APPROVED** [review]

File: cmd/env.go [hunk 3]
```diff
@@ -256,9 +256,7 @@ func getAllEnvGraphqlQuery(appName string) (map[string]string, error) {
 		}
 	`
 
-	var respData struct {
-		GetApp *model.TuberApp
-	}
+	var respData []*model.Tuple
```
[the remark this answers is not in the document]
Comment (jkminneti) [reply cmd/env.go:259]: turns out that broke it tho :( it's reverted and fixed in a subsequent pr

**Pull request merged**